doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,281 @@
+  `OS_VERSION` variable to an integer value.
+  This number must be increased before each build that generates
+  swupd update artefacts.
+
+- Assign a list of bundle names to ``SWUPD_BUNDLES``::
+
+    SWUPD_BUNDLES = "feature_one feature_two"
+
+- For each named bundle, setup a varflag of ``BUNDLE_CONTENTS``
+  that matches the bundle name, and initialize it with a list of
+  packages whose content must be included in the bundle::
+
+    BUNDLE_CONTENTS[feature_one] = "package_one package_three package_six"
+
+Creating a Custom Image and matching Software Update Repository
+===============================================================
+
+Some of the steps listed in this section are optional and cover
+advanced configuration functionality.
+Not all the users will need them, if the default configuration of
+the Ostro OS meets their needs.
+
+The process is exemplified by using real components and describing
+what each step looks like, when applied to such components.
+Unless stated otherwise, the changes described are to be intended
+for the file ``conf/local.conf``
+
+
+Step 0: Review and optionally modify the content of the os-core bundle
+----------------------------------------------------------------------
+The os-core bundle comes with a predefined set of components,
+however each device is likely to have slighlty different requirements.
+In some cases it might be necessary to adjust the set of components
+included in os-core.
+
+To do this, either modify or extend the ``OSTRO_IMAGE_FEATURES_REFERENCE``
+and the ``OSTRO_IMAGE_INSTALL_REFERENCE`` variables for changes
+respectively to the feature and component sets.
+
+Why modifying os-core? Because it's the only unremovable bundle and
+therefore it is well suited to deliver to the device all those tools
+that must always be available.
+
+Example::
+
+  ``OSTRO_IMAGE_INSTALL_REFERENCE += "sudo"``
+
+will add the content of the ``sudo`` recipe to the bundle os-core.
+
+.. note::
+   The step shown works only for adding components to os-core.
+
+   This is the typical case when customizing Ostro OS, because the
+   os-core configuration is very minimal.
+
+   However, in case one needs to remove something from os-core,
+   it is necessary to create either a new image recipe or a new
+   .bbappend file, where to modify the variables that define the
+   content of os-core.
+
+   These are the variable that must be modified::
+
+   - OSTRO_IMAGE_FEATURES_REFERENCE
+   - OSTRO_IMAGE_INSTALL_REFERENCE
+   - IMAGE_INSTALL
+
+   The recommended way is to modify a copy of the original recipe
+   ``ostro-image.bb`` and refer to ``ostro-image.bbclass`` for
+   the values to use.
+
+
+Step 1: Review and optionally modify the list of bundles
+--------------------------------------------------------
+Ostro already defines some bundles, and users can define additional ones.
+
+Example::
+
+  SWUPD_BUNDLES += "sudo_feature"
+  BUNDLE_CONTENTS[sudo_feature] = "sudo"
```

Comment:
I find naming a bundle "feature" misleading, because "feature" is also used in other contexts (image features, distro features) and bundles are (with very few exceptions, see the dev-pkgs feature) explicitly not about those.

I would call this the "sudo_bundle" because it is the bundle which contains sudo. When renaming, also make sure you catch other usages of "feature", not just the actual sudo_feature word.

I understand that this bundle also adds a "feature" to the image (the ability to elevate privileges) but for the sake of this document focusing on "this is a bundle" rather than "this is what the bundle does" is IMHO better.